front/src/applications/opendata/components/Map.js

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import maplibregl from 'maplibre-gl';
+import ReactMapGL, { AttributionControl, ScaleControl } from 'react-map-gl';
+import { point as turfPoint } from '@turf/helpers';
+import { useSelector } from 'react-redux';
+import turfNearestPointOnLine from '@turf/nearest-point-on-line';
+
+/* Main data & layers */
+import Background from 'common/Map/Layers/Background';
+import VirtualLayers from 'applications/osrd/views/OSRDSimulation/VirtualLayers';
+import SnappedMarker from 'common/Map/Layers/SnappedMarker';
+/* Objects & various */
+import TracksGeographic from 'common/Map/Layers/TracksGeographic';
+import colors from 'common/Map/Consts/colors';
+import osmBlankStyle from 'common/Map/Layers/osmBlankStyle';
+import { LAYER_GROUPS_ORDER, LAYERS } from 'config/layerOrder';
+
+import 'common/Map/Map.scss';
+import OperationalPoints from 'common/Map/Layers/OperationalPoints';
+import Platforms from 'common/Map/Layers/Platforms';
+import { getMapMouseEventNearestFeature } from 'utils/mapboxHelper';
+
+export default function Map(props) {
+  const { viewport, setViewport, setClickedFeature } = props;
+  const { mapStyle } = useSelector((state) => state.map);
```

Comment:
Please use a specified selector

Reply:
Specified selector doesn't exists yet for state.map. Should I create it ?

Reply:
or useSelector((state) => state.map.mapStyle at least

Reply:
Resolved with https://github.com/DGEXSolutions/osrd/pull/2586/commits/3df5fac0359358a6c909b0de0830f79114aae25f